include/mbedtls/net_sockets.h

```diff
@@ -106,6 +106,16 @@ mbedtls_net_context;
  */
 void mbedtls_net_init( mbedtls_net_context *ctx );
 
+/**
+ * \brief          Initialize a context with a given file descriptor
+ *
+ * \param ctx      Context to initialize
+ * \param fd       File descriptor to set in the context
+ *
+ * \return         O if succesful, or MBEDTLS_ERR_NET_INVALID_CONTEXT
+ */
+int mbedtls_net_set_fd( mbedtls_net_context *ctx, const int fd );
```

Comment:
Do you want to re-initialize context or just set the fd?
Current name of the function suggests it's only setting the `fd`, however the documentation says it is initializing the context.

Reply:
I'm not usually good when it comes to write documentation, feed back are welcome

Reply:
I tried to rewrite it.